src/pymor/discretizations/iosys.py

```diff
@@ -0,0 +1,1702 @@
+        dB
+            Should the magnitude be in dB on the plot.
+        Hz
+            Should the frequency be in Hz on the plot.
+
+        Returns
+        -------
+        fig
+            Matplotlib figure.
+        ax
+            Matplotlib axes.
+        """
+        assert isinstance(sys_list, InputOutputSystem) or all(isinstance(sys, InputOutputSystem) for sys in sys_list)
+        if isinstance(sys_list, InputOutputSystem):
+            sys_list = (sys_list,)
+
+        assert (plot_style_list is None or isinstance(plot_style_list, str) and len(sys_list) == 1 or
+                all(isinstance(plot_style, str) for plot_style in plot_style_list) and
+                len(sys_list) == len(plot_style_list))
+        if isinstance(plot_style_list, str):
+            plot_style_list = (plot_style_list,)
+
+        import matplotlib.pyplot as plt
+        fig, ax = plt.subplots()
+        for i, sys in enumerate(sys_list):
+            tfw = sys.bode(w)
+            freq = w / (2 * np.pi) if Hz else w
+            mag = spla.norm(tfw, ord=ord, axis=(0, 1))
+            style = '' if plot_style_list is None else plot_style_list[i]
+            if dB:
+                mag = 20 * np.log2(mag)
+                ax.semilogx(freq, mag, style)
+            else:
+                ax.loglog(freq, mag, style)
+        ax.set_title('Magnitude Bode Plot')
+        ax.set_xlabel('Frequency{}'.format(' (Hz)' if Hz else ' (rad/s)'))
+        ax.set_ylabel('Magnitude{}'.format(' (dB)' if dB else ''))
+        return fig, ax
+
+
+_DEFAULT_ME_SOLVER_BACKEND = 'pymess' if config.HAVE_PYMESS else \
+                             'slycot' if config.HAVE_SLYCOT else \
+                             'scipy'
+
+
+class LTISystem(InputOutputSystem):
+    r"""Class for linear time-invariant systems.
+
+    This class describes input-state-output systems given by
+
+    .. math::
+        E x'(t) &= A x(t) + B u(t) \\
+           y(t) &= C x(t) + D u(t)
+
+    if continuous-time, or
+
+    .. math::
+        E x(k + 1) &= A x(k) + B u(k) \\
+          y(k)     &= C x(k) + D u(k)
+
+    if discrete-time, where :math:`A`, :math:`B`, :math:`C`, :math:`D`,
+    and :math:`E` are linear operators.
+
+    Parameters
+    ----------
+    A
+        The |Operator| A.
+    B
+        The |Operator| B.
+    C
+        The |Operator| C.
+    D
+        The |Operator| D or `None` (then D is assumed to be zero).
+    E
+        The |Operator| E or `None` (then E is assumed to be identity).
+    cont_time
+        `True` if the system is continuous-time, otherwise `False`.
+    solver_options
+        The |solver_options| to use to solve the Lyapunov equations.
+    estimator
+        An error estimator for the problem. This can be any object with
+        an `estimate(U, mu, discretization)` method. If `estimator` is
+        not `None`, an `estimate(U, mu)` method is added to the
+        discretization which will call
+        `estimator.estimate(U, mu, self)`.
```

Comment:
Maybe remove this argument? Doesn't seem useful for LTISystem ATM?

Reply:
I made a mistake. It does not seem possible to remove `estimator` and `visualizer`, because they are set in `reduce` methods in `reductors.basic`.

Reply:
Looking at `[Haasdonk, Ohlberger, 2011, Efficient reduced models and a posteriori error estimation for parametrized dynamical systems by offline/online decomposition]`, it does not seem too far fetched to have an estimator for certain LTI systems.

Reply:
I removed the commit.
For the estimator, I guess `LTISystem` class should be first generalized to allow parametric operators?

Reply:
In general, the estimator will depend on the reduction method. (In official pyMOR, we don't have any error estimators for the FOM ATM.)